### DISCUSSION TITLE
[extensions] Composer event emitting

### DISCUSSION THREAD

**Discussion by a-b-r-o-w-n** [microsoft/BotFramework-Composer#5590] [General]
This is a proposal to add event emitting capabilities to Composer core. Extensions (and possibly Composer itself) can register event handlers to execute when certain events are emitted.

Given the right set of core events, this could greatly expand the extensibility of Composer core by allowing extensions to inject custom behavior during the lifecycle of certain Composer actions.

The purpose of this discussion is to gain consensus on the overall design and gather input on what events would be valuable to the extension community.

Here is a [draft PR](https://github.com/microsoft/BotFramework-Composer/pull/5589) that demonstrates what this could look like.

**Comment by a-b-r-o-w-n** [comment]
@peterbozso @joaollq Please take a look at the draft PR and let me know your thoughts on this.

**Reply by peterbozso** [reply]
I've just took a look at it and it looks promising to me. Thanks for sharing this, I am sure we'll be able to take advantage of it in the future!
